blog/_posts/2023-04-20-sbt-plugins-community-repository.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: blog-detail
+post-type: blog
+by: Julien Richard-Foy, Scala Center
+title: "sbt Plugins Community Repository"
+---
+
+On the 7th of April 2023, many Scala builds stopped working because an outage affected the sbt community repository.
+In this blog post, we explain what the sbt community repository is and what it stores, report on the incident,
+and outline what we can do to avoid such problems in the future.
+
+## The Incident
+
+On the 7th of April 2023, many projects of the Scala ecosystem could not anymore resolve
+sbt plugins from the community repository `repo.scala-sbt.org` (examples
+[here](https://github.com/sbt/sbt-dynver/issues/239#issuecomment-1499791434),
+[here](https://github.com/playframework/playframework/issues/11675#issuecomment-1499869916),
+[here](https://github.com/sbt/sbt/issues/7204), and
+[here](https://github.com/spray/sbt-revolver/issues/112)).
+
+## Retrospective
+
+To understand what happened, here is some background information. sbt plugins are packaged
+as modules published to Ivy or Maven repositories. Historically, sbt plugins used to be
+published primarily to the sbt community repository hosted on Bintray. However, in 2021,
+JFrog has [sunsetted the Bintray
+service](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/).
+Since then, JFrog has granted the Scala Center a cloud-hosted Artifactory instance for free.
+As it was explained in the [release notes of sbt
+1.5.1](https://eed3si9n.com/bintray-to-jfrog-artifactory-migration-status-and-sbt-1.5.1),
+this Artifactory instance has been used as a read-only repository to host all the sbt
+plugins that were published to the community repository at that time, but the recommendation
+to plugin authors was to switch to the Central Maven repository.
```

Comment:
I'm pretty sure it's called the Maven Central repository, not Central Maven.
```suggestion
to plugin authors was to switch to the Maven Central repository.
```
Same in several places in the post.

Reply:
I think it’s called just “Central”. And it’s a Maven repository, hence “the Central Maven repository”. It could be abbreviated to “the Central repository”, but I left “Maven” for clarity. See https://central.sonatype.org/

Reply:
In practice, everybody calls it “Maven Central”. I'm not sure if I've ever heard anyone call it anything else — I don't think I have.

Reply:
OK, I’ve changed that.